pkg/cli/image/extract/extract.go

```diff
@@ -387,10 +387,13 @@ func (o *ExtractOptions) Run() error {
 						alter = append(alter, newCopyFromDirectory(mapping.From))
 					default:
 						name, parent := path.Base(mapping.From), path.Dir(mapping.From)
-						if name == "." || parent == "." {
+						if name != "." && parent == "." {
+							alter = append(alter, newCopyFromRootDir(name))
```

Comment:
you _could_ get away with adding a RootDir boolean to copyFromPattern struct, I experimented with doing that - that saves needing the separate Alter, copyFromRootDir. I pushed the branch w/ that wdyt?  https://github.com/zerodayz/oc/compare/FixImageExtract-From-RootDir...sallyom:zerodayz-FixImageExtract-From-RootDir?expand=1

Reply:
I left a code I used for debugging there, removed and I tested your code updates, it works very well. I think bug fixed :) Thank you @sallyom here is new commit for the review: https://github.com/openshift/oc/commit/03aff152605399a5ca06ca342b2d2ead730a572c